themes/qgrid/source/js/menu.js

```diff
@@ -15,46 +17,233 @@ function toggleVisibility(e) {
 	}
 }
 
-function filter(e) {
+function onSearchChange(e) {
 	e.stopImmediatePropagation();
 	const input = e.target || e.srcElement;
-	const filter = input.value.toLowerCase();
+	const search = input.value.toLowerCase();
+	setSearch(search);
+}
+
+function searchOnLoad() {
+	const active = document.querySelector('.active-topic');
+	const search = getSearch();
+	if (active && search) {
+		document.getElementById('search').value = search;
+		updateMenuLinks(search);
+		currentPageSearch(search);
+		menuItemsSearch(search);
+	}
+}
+
+function searchOnHashChange() {
+	const search = getSearch();
+	document.getElementById('search').value = search;
+	updateMenuLinks(search);
+	currentPageSearch(search);
+	menuItemsSearch(search);
+}
+
+function currentPageSearch(search) {
+	let scrolled = false;
+	document.querySelectorAll('.page-wrapper *').forEach(node => {
+		if (node.nodeName == 'H2' || node.nodeName == 'H3' || node.nodeName == 'P') {
+			const a = node.querySelector('a');
+			const pageItem = (a && a.textContent) ? a : node;
+			const { textContent } = pageItem;
+			if (isCorrectSearch(textContent, search)) {
```

Comment:
rename to testSearch, isCorrect doesnt make sense.

Reply:
we have isCorrectSearch and highlightText something wrong here... also isCorrectSearch look the same to below logic